### PR TITLE
chore: Add README file to explain xTests

### LINKED ISSUE #11680: Document `XTest` concepts, best practices, and TODOs
### Problem

The `XTest` framework now has many examples (of varying clarity), but no documentation.

### Solution

Add a README clarifying what an `XTest` is, highlight the best examples, and list issues with the current state of affairs.

### Alternatives

_No response_

### PR DESCRIPTION
**Pull request by lukelee-sl**
**Description**:
Add a README file to the xTests directory to help motivate its purpose and provide a general description of how and when they should be added.

Fixes #11680

### CODE REVIEW

File: hedera-node/hedera-app/src/xtest/java/README.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+## Introduction
+XTests are intended to fill the gap between unit tests and integration tests. 
+They are designed to test the functionality of a services module in its entirety, as well as 
+interactions between module.
+
+## Goals and Benefits
+- Easier to write, read and understand, and to debug into than the HAPI suite BDDs because they just exercise the service (or services directly) through its API instead of being an end-to-end test where a HAPI transaction goes in and a record stream comes out, with all kinds of concurrency and asynchronicity happening in between, and the measurement of the effect being rather indirect.
+- At the same time making it easier to test edge cases because you can hand-craft test data using the actual data structures taken by the API, and can thus drive the system into states it can't normally get to by using only valid HAPI transactions.
+- A more complete test of the service-under-test than (our current) unit tests because mocking is extremely discouraged. So you're actually testing the real code instead of hard-wiring in to the test a bunch of assumptions of how the real code is supposed to work.
```
Comment (david-bakin-sl) [inline hedera-node/hedera-app/src/xtest/java/README.md:11]: to clarify: "because mocking is extremely discouraged" -> "because mocking in XTests is extremely discouraged" ? (just in case anyone hasn't figured out it is the _unit tests_ that are full of mocks)

File: hedera-node/hedera-app/src/xtest/java/README.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+- xTests executes more quickly and thus allows the developer to iterate between test and functionality more efficiently than integration tests
+
+## When xTests Are Not Appropriate
+
+While xTests offer benefits as described above, they are not appropriate for all testing scenarios.
+Some specific scenarios where xTests are not appropriate include:
+
+- Any test that require multiple nodes to be running
+- Tests that require complex signing/signature verification
+- Any test where the existing scaffolding infrastructure needs to be significantly modified (although this will likely become more flexible over time)
```
Comment (david-bakin-sl) [inline hedera-node/hedera-app/src/xtest/java/README.md:23]: "flexible" -> "flexible/powerful" ?

**Review by david-bakin-sl — APPROVED** [review]
This is fine for now (and about all that can be said for now).  But let's remember to revisit this as we write new XTests and build the XTest framework.

**Review by Neeharika-Sompalli — APPROVED** [review]
LGTM ! Thanks @lukelee-sl
